### PR TITLE
(MODULES-6708) fix tests for windows agent to agent upgrades

### PR DESCRIPTION
**Pull request by speedofdark**
* remove tests around configuration migration as they are not managed by the upgrade
* changed base version from 3.x to 1.7.0 and associated changes
* specified upgrade target as 1.10.0

Three Windows tests are currently red. Two appear to be legitimate failues related to
idempotency. The third is related to service startup.

Tests have only been updated to make things green, with no other refactoring applied
as yet.

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -23,9 +23,17 @@ class { 'puppet_agent': }
...
     end
 
-    describe service('puppet') do
-      it { is_expected.to be_enabled }
-      it { is_expected.to be_running }
+    if default['platform'] =~ /windows/i
+      # puppet service is not started automagically on windows
+      describe service('puppet') do
+        it { is_expected.to be_enabled }
+        it { is_expected.to_not be_running }
```
Comment (glennsarti) [inline spec/acceptance/class_spec.rb]: This appears very odd....Not sure we should be expecting this.

**Comment by speedofdark** [comment]
We reverted the change that Glenn commented on above; this results in a total of 3 red tests for now. This will be addressed separately.

**Comment by speedofdark** [comment]
1.7.0 was the version shipped in the 2016.4.0 (the initial LTS release), and the module doesn't work well for 5 yet on every OS we support so we use 1.10.0 as the last of the 4 series. We could pick 1.10.12 instead. It doesn't really matter (shrug).

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -135,24 +134,19 @@ def setup_puppet_on(host, opts = {})
 
   puts "Setup foss puppet on #{host}"
   configure_defaults_on host, 'foss'
-  install_puppet_on host, :version => ENV['PUPPET_CLIENT_VERSION'] || '3.8.6'
+  install_puppet_agent_on host, :version => ENV['PUPPET_CLIENT_VERSION'] || '1.7.0'
```
Comment (glennsarti) [inline spec/spec_helper_acceptance.rb]: 👍

**Pull request merged**